lib/net.ts

```diff
@@ -0,0 +1,145 @@
+/**
+ * Net - abstraction layer around Node's HTTP/S request system.
+ * Advantages:
+ * - easier acquiring of data
+ * - mass disabling of outgoing requests via Config.
+ */
+
+import * as https from 'https';
+import * as http from 'http';
+import * as url from 'url';
+import * as Streams from './streams';
+import { resolve } from 'dns';
+
+export interface NetRequestOptions extends https.RequestOptions {
+	body?: string;
+	query?: {[k: string]: string | number};
+}
+
+export class NetStream extends Streams.ReadWriteStream {
+	statusCode?: number;
+	headers?: http.IncomingHttpHeaders;
+	opts?: NetRequestOptions;
+	uri: string;
+	request: http.ClientRequest | null;
+	responseStream: Streams.ReadStream | null;
+	constructor(uri: string, opts?: NetRequestOptions | null) {
+		super();
+		this.statusCode = 0;
+		this.headers = {};
+		this.opts = opts ? opts : undefined;
+		this.uri = uri
+		this.responseStream = null;
+		// make request
+		this.request = null;
+		this.makeRequest();
+	}
+	makeRequest() {
+		const protocol = url.parse(this.uri).protocol as string;
+		const net = protocol.includes('https:') ? https : http;
+		return new Promise(resolve => {
+			 this.request = net.get(this.opts ? this.opts : this.uri, response => {
+				  response.setEncoding('utf-8');
+				  const stream = new Streams.ReadStream({nodeStream: response});
```

Comment:
My idea was to make the new stream _be_ the stream, not _contain_ the stream.

Reply:
Alright, i'll rewrite it.